lib_eio_linux/eio_linux.ml

```diff
@@ -1207,6 +1208,42 @@ class dir ~label (fd : dir_fd) = object
     | Dir_fd -> Some fd
     | _ -> None
 
+  method stat path =
+    Switch.run @@ fun sw ->
+    let fd = Low_level.openat ~sw ~seekable:false fd path
+          ~access:`R
+          ~flags:Uring.Open_flags.(path + cloexec + nofollow)
```

Comment:
So `nofollow` makes this behave as `lstat`? Needs a comment I think, and maybe a separate function. Otherwise, `File.load` on a symlink will use the size of the link instead of the size of the target, won't it? Could do with a test-case, anyway.

Reply:
Pretty much, I added this late when I was writing the symbolic link test. Now thinking about it, we should probably make `stat` get an optional argument for `nofollow`, `load` would want it to `follow` while direct calls would probably want it to `nofollow`. Also, should we restrict `load` just to a resulting `Regular_file` ? Can't think of any more uses.